packages/libs/eda/src/lib/map/analysis/MapAnalysis.tsx

```diff
@@ -772,11 +774,20 @@ function MapAnalysisImpl(props: ImplProps) {
     </div>
   );
 
-  // close left-side panel when map events happen
-  const closePanel = useCallback(
-    () => setIsSidePanelExpanded(false),
-    [setIsSidePanelExpanded]
-  );
+  // deselect selectedMarkers and close panel
+  const deselectMarkers = useCallback(() => {
+    updateMarkerConfigurations({
+      // type assertion to avoid ts error
+      ...(activeMarkerConfiguration as PieMarkerConfiguration),
+      selectedMarkers: undefined,
+    });
+    // close side panel
+    setIsSidePanelExpanded(false);
+  }, [
+    updateMarkerConfigurations,
+    activeMarkerConfiguration,
+    setIsSidePanelExpanded,
+  ]);
```

Comment:
This seems to remove the arbitrary choice of `PieMarkerConfiguration` & function renaming removes need for comments. Don't forget to remove unused imports.

```suggestion
  const deselectMarkersAndClosePanel = useCallback(() => {
    updateMarkerConfigurations({
      ...(activeMarkerConfiguration as MarkerConfiguration),
      selectedMarkers: undefined,
    });
    setIsSidePanelExpanded(false);
  }, [
    updateMarkerConfigurations,
    activeMarkerConfiguration,
    setIsSidePanelExpanded,
  ]);
```

Reply:
Thank you for your valuable comment @bobular 👍 I missed MarkerConfiguration 😅 Will comply with your comments and then merge